api/core/v1alpha/domain_types.go

```diff
@@ -60,15 +60,18 @@ type DomainSpec struct {
 	// (e.g. an EdgeFunction or a TunnelEndpoint).
 	// This is a Pro feature only.
 	ForwardingSpec *DomainForwardingSpec `json:"forwarding,omitempty"`
+
+	// EdgeFunction filters applied for the domain.
+	Filters []*LocalObjectReference `json:"filters,omitempty"`
 }
 
 type DomainTargetSpec struct {
 	// Represents targets specified via DNS.
 	DNS *DomainTargetDNS `json:"dns,omitempty"`
 
 	// Represent a target specified via a reference to another object
-	// within Apoxy (e.g. Proxy, EdgeFunction, TunnelEndpoint).
-	Ref *DomainTargetRef `json:"ref,omitempty"`
+	// within Apoxy (e.g. Proxy, EdgeFunction (type=backend), TunnelEndpoint).
```

Comment:
this makes me wonder if we should have a "FilterFunction" and "BackendFunction" type instead?

Reply:
Not a bad idea. I'll throw it on a backlog.